spec/models/season/phase_switcher_spec.rb

```diff
@@ -1,35 +1,43 @@
 require 'spec_helper'
 
+describe Season::PhaseSwitcher do
 
-context 'when switching phases' do
-  subject { Season.current }
+  it "is available as described class" do
+    expect(described_class).to eq Season::PhaseSwitcher
+  end
+
+  context 'when switching phases' do
+    season = Season.current
 
-  describe '#fake_application_phase' do
-    it 'timeshifts the application phase to today' do
-      Timecop.travel(Season.current.applications_close_at - 1.month) do
-        Season::PhaseSwitcher.fake_application_phase
-        expect(subject).to be_application_period
+    context 'when the input is valid' do
+      it 'gets called on the season object' do
+        phase = :fake_application_phase
+        described_class.destined(phase)
+        season.reload
+        expect(season).to be_application_period
       end
-    end
-  end
 
-  describe '#fake_coding_phase' do
-    it 'timeshifts the coding phase to today' do
-      Timecop.travel(Season.current.ends_at - 1.week) do
-        Season::PhaseSwitcher.fake_coding_phase
-        expect(subject).to be_started
+      it 'changes also when real time happens to be within application_period' do
+        phase = :fake_proposals_phase
+        described_class.destined(phase)
+        season.reload
+        expect(season).not_to be_application_period
       end
     end
-  end
 
-  describe '#fake_proposals_phase' do
-    it 'timeshifts the proposal period to today' do
-      Timecop.travel(Season.current.project_proposals_close_at - 2.weeks) do
-        fake_time = Time.now #as returned by Timecop
-        Season::PhaseSwitcher.fake_proposals_phase
-        expect(subject.project_proposals_open_at).to be < fake_time
-        expect(subject.project_proposals_close_at).to be > fake_time
+    context 'when the input is malicious' do
+      phase = :bad_intentions
+
+      it 'raises an error when it receives a non-whitelisted phase' do
+        expect {
+          described_class.destined(phase)
+        }.to raise_error(ArgumentError)
+      end
+      it 'does not change the season dates' do
```

Comment:
I'm not a huge fan of this `rescue` pattern. How about using a chained expectation instead?
Should look something along the lines of this if I recall correctly:
```ruby
RSpec::Matchers.define_negated_matcher :not_change, :change

expect { described_class.destined(phase) }
  .to raise_error(ArgumentError)
  .and not_change { Season.current.reload }
```